classes/manager.php

```diff
@@ -163,12 +168,190 @@ public static function get_reasons(float $duration): int {
         if (isset($SESSION->toolexcimerflameall)) {
             $reason |= self::REASON_FLAMEALL;
         }
-        if (($duration * 1000) >= (int) get_config('tool_excimer', 'trigger_ms')) {
+
+        if (($duration * 1000) >= self::get_min_duration_for_reason_slow()) {
             $reason |= self::REASON_AUTO;
         }
         return $reason;
     }
 
+    /**
+     * Returns the minimum duration for profiles matching this reason and page/request.
+     *
+     * @param  int $reason - the profile type or REASON_*
+     * @return float duration (in milliseconds) of the fastest profile for a given reason and request/page.
+     */
+    public static function get_min_duration_for_request_and_reason(string $request, int $reason): float {
```

Comment:
At the moment there is only one reason (slow) for a given url. But lets say there ends up being 10, wouldn't this be better if there was a single cache item for a given request and all the metadata for a url was in that cache item rather than doing 10 cache reads? Some of the things we might stash in here might not be timings so I'd call it something a bit more generic eg: make('tool_excimer', 'request_metadata');

Reply:
Yes that makes sense.